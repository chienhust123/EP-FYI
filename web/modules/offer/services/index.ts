import axios from 'axios';
import { searchAggregatedCompanyStatsList } from '@/services/offer';

export const searchOffersByCompany = (searchValue: string, offset: number, limit: number) =>
  searchAggregatedCompanyStatsList({
    offset,
    limit,
    filter_options: {
      company_name_query: searchValue,
    },
  }).then((resp) => {
    return resp.company_stat_list.map((company) => ({
      value: company.company.id.toString(),
      label: company.company.name,
    }));
  });

export const searchOffersByCountry = (searchValue: string, offset: number, limit: number) =>
  searchAggregatedCompanyStatsList({
    offset,
    limit,
    filter_options: {
      country_list: [searchValue],
    },
  }).then((resp) =>
    resp.company_stat_list.map((company) => ({
      value: company.company.id.toString(),
      label: company.company.name,
    }))
  );
