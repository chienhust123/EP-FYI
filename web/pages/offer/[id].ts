import { DetailPage } from '@/modules/offer/pages/Detail';

export default DetailPage;
