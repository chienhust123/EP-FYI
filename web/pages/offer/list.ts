import { SearchPage } from '@/modules/offer/pages/Search';

export default SearchPage;
