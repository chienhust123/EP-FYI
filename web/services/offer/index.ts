export * from "./models/index";
export * from "./http/index";