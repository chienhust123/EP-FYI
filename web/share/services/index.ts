import axios from 'axios';

export const uploadImageToPresignedUrl = async (presignedUrl: string, selectedFile: File) => {
  const resp = await axios.put(presignedUrl, selectedFile, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  });

  return resp.data;
};
