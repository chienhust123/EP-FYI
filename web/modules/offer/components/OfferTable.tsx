import React from 'react';
import { Table, ActionIcon, Box, Pagination, Flex } from '@mantine/core';
import { Offer } from '@/services/offer';
import { formatCurrency } from '@/share/helpers/format-currency';
import Link from 'next/link';
import { IconEye } from '@tabler/icons-react';
import { OfferStatusMap } from '@/share/consts';

type Props = {
  offerList: Offer[];
}

const OfferTable: React.FC<Props> = ({ offerList }) => {

  const rows = offerList.map((offer) => (
    <Table.Tr
      key={offer.id}
    >
      <Table.Td>{offer.company.name}</Table.Td>
      <Table.Td>{offer.location.city}</Table.Td>
      <Table.Td>{offer.position.title}</Table.Td>
      <Table.Td> {formatCurrency(offer.total_package)}</Table.Td>
      <Table.Td>{OfferStatusMap[offer.status]}</Table.Td>
      <Table.Td>
        <Link href={`/offer/detail?id=${offer.id}`}>
          <ActionIcon variant='outline' >
            <IconEye />
          </ActionIcon>
        </Link>
      </Table.Td>
    </Table.Tr>
  ));

  return (
    <Box w={1000}>
      <Table>
        <Table.Thead>
          <Table.Tr>
            <Table.Th>Company</Table.Th>
            <Table.Th>Location</Table.Th>
            <Table.Th>Position</Table.Th>
            <Table.Th>Total Package</Table.Th>
            <Table.Th>Status</Table.Th>
            <Table.Th />
          </Table.Tr>
        </Table.Thead>
        <Table.Tbody>{rows}</Table.Tbody>
      </Table>
      <Flex mt='lg' justify='end'>
        <Pagination total={10} />
      </Flex>
    </Box>
  );
}

export default OfferTable;