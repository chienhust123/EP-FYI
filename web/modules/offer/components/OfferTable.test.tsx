import { screen } from '@testing-library/react';
import OfferTable from './OfferTable';
import { Offer, OfferStatus, PositionLevel } from '@/services/offer';
import { renderApp } from '@/test/render';

describe('OfferTable', () => {
  const mockOffers: Offer[] = [
    {
      id: 1,
      company: {
        id: 101,
        name: 'Tech Innovators',
        profile_image_url: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAPAAAADSCAMAAABD772dAAAAw1BMVEX///95590uWrUAydM5jf+B6N/X+PUrWLTN1+1CaLv5/v72+v+Etf+M4+j0/f2wz/9v3eMwiv8iU7IAy9Tz+f/t+/p/6N6g7udy5tvI9fGW7OTw/Pvi6fVSdMClt96uvuHQ8/ZN1t3i+fq77vJlpv88kf+Yq9iPu//R5P+z8evN9vGa7eW78u1hgcaIoNS4x+Zwjcs3Yrnk6vVpiMhBZ7qi6eyI4+hl2uGdxf/c6/9Mmf+40/9zq//A2/+lyP97ls/I0uoYrnA8AAAEhklEQVR4nO3dbVeyQBAGYCEBtcKkVNIsw7QsrSzrKXv9/7/qmRXsWMnKW6ws9/25c9zrzLDzqdlCQYroraFer4g+RYrRG7aiNUd63RR9kpSiawqL1moe6nXRh0kjHlhR7Dnakb67v8Cu2tYIbcqs/g722nt46bRlRa8AU3cr2hWhpTSvArvoBlVab4s+X+LxA7tftK2wMS3VN80DL25vb0xviz5rIlkLnqNpYo0K4+ubI9HHjZ9A4HkKlmUZHUIfZLrDQ4BVVTUMyyjfDggt+tyREw7smSedj8FnRts7NNhVW+qkc3tcyuA9Fg1MZEJb1oTQd9lSRwUvKm2pneuTLKHjgb32Lo8JnZF7LD7YvcfUKU2sO9GaAEkC7KKNxZgWTeInKfBXqcv3xxtd6ETBrnlaEo3iJWkwkXMHLgO8SQEYYIABBlg0iheAAQYYYIBFo3gBGGCAAQZYNIoXgAEGGGCARaN4ARhggAEGWDSKF4ABBhhggEWjeAEYYIABBliwqdLu7p4++PyzkWxg0+k+7O1vbW3t5gFM2MdTps0FWH/YO933tJKDK6bDsFvfIiu4Und6jz+x0oLNbm/33yqtlOB279EPKx14Pmj9rXKBvwbtmkgCrvR+Xcdyg829QFqAAQYYYIABBhhggAEGGGCANxicwMKeTIG3n3aqzzG3rWULvFOr1WaEPohe6qyBi0Uy98/OX57vopU6e2CWWq04eyV0hDVF2QS75v7s9ektrDmzYE/N2vstzLq1bIM99Gzn/S1oe2cfvCg0oYOs4ZICvPimz2hi5QbsVZqN6QvemJYK7KnpHvMvtHzgubmaL3CxCDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwzw6nR91w1LCvZ2LK83ywMucLZoywp243AX80oIplTma/9XquUEs9SX3jrIBZhiEvrnPSY12FXTxNqsdx7+GMxS7+0uLrJ8gAtsh7w7ph/zAi6wRwJoTG/AazypgbkBGGCAAQYYYLnB05yBrUl+wIZhWZPxwGdfo1xgQyWsNf04KfkuIpUHbFDUSefj5Ii7dFUSMLWxMe18HH+uXTwqBdiyVIbll1YOMLugLHU8+Ay8PDgguMbbaykKTG08Kd8PPkOthw4Apj9g+8UjbNoOnPBgNno6t4Obo7BLz9eA2V7xs/PqxV9qC2HB8xvqfnBTinIq/vbhPsMGugviJQSYvlmrc31zFLUEPmC2dLj/+n4RZpF4jIQAjwkb56dWgN1V8e9/3cbLCQq2ldgPcfwEU2FnO09va5doJ5sgYFtpaFfD2D/14xUA9ghAlJcPYmY92CbspdOO/1MLMPtoz84JG/MZk2jhg21Fa146ZjInY2Bv0KbcxsvhgLVW87Ke4E8ROI1BuyY+YK01PEyijZez/VJNa/Zw8htsK63moV4X8oGlkO9g225cjfS2rFiWLzDdT1rz0JHZOo8HpgtqJG8bL0dvsNEz0h1T9ElSit4aBqzsfw2lSZ9iE2euAAAAAElFTkSuQmCC',
      },
      location: {
        id: 201,
        country: 'US',
        state: 'CA',
        city: 'San Francisco',
      },
      position: {
        id: 301,
        title: 'Software Engineer',
        level: PositionLevel.VALUE_MIDDLE,
      },
      total_package: {
        amount: 120000,
        currency: 'USD',
      },
      image_url: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAPAAAADSCAMAAABD772dAAAAw1BMVEX///95590uWrUAydM5jf+B6N/X+PUrWLTN1+1CaLv5/v72+v+Etf+M4+j0/f2wz/9v3eMwiv8iU7IAy9Tz+f/t+/p/6N6g7udy5tvI9fGW7OTw/Pvi6fVSdMClt96uvuHQ8/ZN1t3i+fq77vJlpv88kf+Yq9iPu//R5P+z8evN9vGa7eW78u1hgcaIoNS4x+Zwjcs3Yrnk6vVpiMhBZ7qi6eyI4+hl2uGdxf/c6/9Mmf+40/9zq//A2/+lyP97ls/I0uoYrnA8AAAEhklEQVR4nO3dbVeyQBAGYCEBtcKkVNIsw7QsrSzrKXv9/7/qmRXsWMnKW6ws9/25c9zrzLDzqdlCQYroraFer4g+RYrRG7aiNUd63RR9kpSiawqL1moe6nXRh0kjHlhR7Dnakb67v8Cu2tYIbcqs/g722nt46bRlRa8AU3cr2hWhpTSvArvoBlVab4s+X+LxA7tftK2wMS3VN80DL25vb0xviz5rIlkLnqNpYo0K4+ubI9HHjZ9A4HkKlmUZHUIfZLrDQ4BVVTUMyyjfDggt+tyREw7smSedj8FnRts7NNhVW+qkc3tcyuA9Fg1MZEJb1oTQd9lSRwUvKm2pneuTLKHjgb32Lo8JnZF7LD7YvcfUKU2sO9GaAEkC7KKNxZgWTeInKfBXqcv3xxtd6ETBrnlaEo3iJWkwkXMHLgO8SQEYYIABBlg0iheAAQYYYIBFo3gBGGCAAQZYNIoXgAEGGGCARaN4ARhggAEGWDSKF4ABBhhggEWjeAEYYIABBliwqdLu7p4++PyzkWxg0+k+7O1vbW3t5gFM2MdTps0FWH/YO933tJKDK6bDsFvfIiu4Und6jz+x0oLNbm/33yqtlOB279EPKx14Pmj9rXKBvwbtmkgCrvR+Xcdyg829QFqAAQYYYIABBhhggAEGGGCANxicwMKeTIG3n3aqzzG3rWULvFOr1WaEPohe6qyBi0Uy98/OX57vopU6e2CWWq04eyV0hDVF2QS75v7s9ektrDmzYE/N2vstzLq1bIM99Gzn/S1oe2cfvCg0oYOs4ZICvPimz2hi5QbsVZqN6QvemJYK7KnpHvMvtHzgubmaL3CxCDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwzw6nR91w1LCvZ2LK83ywMucLZoywp243AX80oIplTma/9XquUEs9SX3jrIBZhiEvrnPSY12FXTxNqsdx7+GMxS7+0uLrJ8gAtsh7w7ph/zAi6wRwJoTG/AazypgbkBGGCAAQYYYLnB05yBrUl+wIZhWZPxwGdfo1xgQyWsNf04KfkuIpUHbFDUSefj5Ii7dFUSMLWxMe18HH+uXTwqBdiyVIbll1YOMLugLHU8+Ay8PDgguMbbaykKTG08Kd8PPkOthw4Apj9g+8UjbNoOnPBgNno6t4Obo7BLz9eA2V7xs/PqxV9qC2HB8xvqfnBTinIq/vbhPsMGugviJQSYvlmrc31zFLUEPmC2dLj/+n4RZpF4jIQAjwkb56dWgN1V8e9/3cbLCQq2ldgPcfwEU2FnO09va5doJ5sgYFtpaFfD2D/14xUA9ghAlJcPYmY92CbspdOO/1MLMPtoz84JG/MZk2jhg21Fa146ZjInY2Bv0KbcxsvhgLVW87Ke4E8ROI1BuyY+YK01PEyijZez/VJNa/Zw8htsK63moV4X8oGlkO9g225cjfS2rFiWLzDdT1rz0JHZOo8HpgtqJG8bL0dvsNEz0h1T9ElSit4aBqzsfw2lSZ9iE2euAAAAAElFTkSuQmCC',
      status: 1,
      created_time: 1627891200,
    },
    {
      id: 2,
      company: {
        id: 102,
        name: 'Global Finance',
        profile_image_url: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAPAAAADSCAMAAABD772dAAAAw1BMVEX///95590uWrUAydM5jf+B6N/X+PUrWLTN1+1CaLv5/v72+v+Etf+M4+j0/f2wz/9v3eMwiv8iU7IAy9Tz+f/t+/p/6N6g7udy5tvI9fGW7OTw/Pvi6fVSdMClt96uvuHQ8/ZN1t3i+fq77vJlpv88kf+Yq9iPu//R5P+z8evN9vGa7eW78u1hgcaIoNS4x+Zwjcs3Yrnk6vVpiMhBZ7qi6eyI4+hl2uGdxf/c6/9Mmf+40/9zq//A2/+lyP97ls/I0uoYrnA8AAAEhklEQVR4nO3dbVeyQBAGYCEBtcKkVNIsw7QsrSzrKXv9/7/qmRXsWMnKW6ws9/25c9zrzLDzqdlCQYroraFer4g+RYrRG7aiNUd63RR9kpSiawqL1moe6nXRh0kjHlhR7Dnakb67v8Cu2tYIbcqs/g722nt46bRlRa8AU3cr2hWhpTSvArvoBlVab4s+X+LxA7tftK2wMS3VN80DL25vb0xviz5rIlkLnqNpYo0K4+ubI9HHjZ9A4HkKlmUZHUIfZLrDQ4BVVTUMyyjfDggt+tyREw7smSedj8FnRts7NNhVW+qkc3tcyuA9Fg1MZEJb1oTQd9lSRwUvKm2pneuTLKHjgb32Lo8JnZF7LD7YvcfUKU2sO9GaAEkC7KKNxZgWTeInKfBXqcv3xxtd6ETBrnlaEo3iJWkwkXMHLgO8SQEYYIABBlg0iheAAQYYYIBFo3gBGGCAAQZYNIoXgAEGGGCARaN4ARhggAEGWDSKF4ABBhhggEWjeAEYYIABBliwqdLu7p4++PyzkWxg0+k+7O1vbW3t5gFM2MdTps0FWH/YO933tJKDK6bDsFvfIiu4Und6jz+x0oLNbm/33yqtlOB279EPKx14Pmj9rXKBvwbtmkgCrvR+Xcdyg829QFqAAQYYYIABBhhggAEGGGCANxicwMKeTIG3n3aqzzG3rWULvFOr1WaEPohe6qyBi0Uy98/OX57vopU6e2CWWq04eyV0hDVF2QS75v7s9ektrDmzYE/N2vstzLq1bIM99Gzn/S1oe2cfvCg0oYOs4ZICvPimz2hi5QbsVZqN6QvemJYK7KnpHvMvtHzgubmaL3CxCDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwzw6nR91w1LCvZ2LK83ywMucLZoywp243AX80oIplTma/9XquUEs9SX3jrIBZhiEvrnPSY12FXTxNqsdx7+GMxS7+0uLrJ8gAtsh7w7ph/zAi6wRwJoTG/AazypgbkBGGCAAQYYYLnB05yBrUl+wIZhWZPxwGdfo1xgQyWsNf04KfkuIpUHbFDUSefj5Ii7dFUSMLWxMe18HH+uXTwqBdiyVIbll1YOMLugLHU8+Ay8PDgguMbbaykKTG08Kd8PPkOthw4Apj9g+8UjbNoOnPBgNno6t4Obo7BLz9eA2V7xs/PqxV9qC2HB8xvqfnBTinIq/vbhPsMGugviJQSYvlmrc31zFLUEPmC2dLj/+n4RZpF4jIQAjwkb56dWgN1V8e9/3cbLCQq2ldgPcfwEU2FnO09va5doJ5sgYFtpaFfD2D/14xUA9ghAlJcPYmY92CbspdOO/1MLMPtoz84JG/MZk2jhg21Fa146ZjInY2Bv0KbcxsvhgLVW87Ke4E8ROI1BuyY+YK01PEyijZez/VJNa/Zw8htsK63moV4X8oGlkO9g225cjfS2rFiWLzDdT1rz0JHZOo8HpgtqJG8bL0dvsNEz0h1T9ElSit4aBqzsfw2lSZ9iE2euAAAAAElFTkSuQmCC',
      },
      location: {
        id: 202,
        country: 'GB',
        state: '',
        city: 'London',
      },
      position: {
        id: 302,
        title: 'Financial Analyst',
        level: PositionLevel.VALUE_PRINCIPAL,
      },
      total_package: {
        amount: 70000,
        currency: 'GBP',
      },
      image_url: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAPAAAADSCAMAAABD772dAAAAw1BMVEX///95590uWrUAydM5jf+B6N/X+PUrWLTN1+1CaLv5/v72+v+Etf+M4+j0/f2wz/9v3eMwiv8iU7IAy9Tz+f/t+/p/6N6g7udy5tvI9fGW7OTw/Pvi6fVSdMClt96uvuHQ8/ZN1t3i+fq77vJlpv88kf+Yq9iPu//R5P+z8evN9vGa7eW78u1hgcaIoNS4x+Zwjcs3Yrnk6vVpiMhBZ7qi6eyI4+hl2uGdxf/c6/9Mmf+40/9zq//A2/+lyP97ls/I0uoYrnA8AAAEhklEQVR4nO3dbVeyQBAGYCEBtcKkVNIsw7QsrSzrKXv9/7/qmRXsWMnKW6ws9/25c9zrzLDzqdlCQYroraFer4g+RYrRG7aiNUd63RR9kpSiawqL1moe6nXRh0kjHlhR7Dnakb67v8Cu2tYIbcqs/g722nt46bRlRa8AU3cr2hWhpTSvArvoBlVab4s+X+LxA7tftK2wMS3VN80DL25vb0xviz5rIlkLnqNpYo0K4+ubI9HHjZ9A4HkKlmUZHUIfZLrDQ4BVVTUMyyjfDggt+tyREw7smSedj8FnRts7NNhVW+qkc3tcyuA9Fg1MZEJb1oTQd9lSRwUvKm2pneuTLKHjgb32Lo8JnZF7LD7YvcfUKU2sO9GaAEkC7KKNxZgWTeInKfBXqcv3xxtd6ETBrnlaEo3iJWkwkXMHLgO8SQEYYIABBlg0iheAAQYYYIBFo3gBGGCAAQZYNIoXgAEGGGCARaN4ARhggAEGWDSKF4ABBhhggEWjeAEYYIABBliwqdLu7p4++PyzkWxg0+k+7O1vbW3t5gFM2MdTps0FWH/YO933tJKDK6bDsFvfIiu4Und6jz+x0oLNbm/33yqtlOB279EPKx14Pmj9rXKBvwbtmkgCrvR+Xcdyg829QFqAAQYYYIABBhhggAEGGGCANxicwMKeTIG3n3aqzzG3rWULvFOr1WaEPohe6qyBi0Uy98/OX57vopU6e2CWWq04eyV0hDVF2QS75v7s9ektrDmzYE/N2vstzLq1bIM99Gzn/S1oe2cfvCg0oYOs4ZICvPimz2hi5QbsVZqN6QvemJYK7KnpHvMvtHzgubmaL3CxCDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwwwwAADDDDAAAMMMMAAAwzw6nR91w1LCvZ2LK83ywMucLZoywp243AX80oIplTma/9XquUEs9SX3jrIBZhiEvrnPSY12FXTxNqsdx7+GMxS7+0uLrJ8gAtsh7w7ph/zAi6wRwJoTG/AazypgbkBGGCAAQYYYLnB05yBrUl+wIZhWZPxwGdfo1xgQyWsNf04KfkuIpUHbFDUSefj5Ii7dFUSMLWxMe18HH+uXTwqBdiyVIbll1YOMLugLHU8+Ay8PDgguMbbaykKTG08Kd8PPkOthw4Apj9g+8UjbNoOnPBgNno6t4Obo7BLz9eA2V7xs/PqxV9qC2HB8xvqfnBTinIq/vbhPsMGugviJQSYvlmrc31zFLUEPmC2dLj/+n4RZpF4jIQAjwkb56dWgN1V8e9/3cbLCQq2ldgPcfwEU2FnO09va5doJ5sgYFtpaFfD2D/14xUA9ghAlJcPYmY92CbspdOO/1MLMPtoz84JG/MZk2jhg21Fa146ZjInY2Bv0KbcxsvhgLVW87Ke4E8ROI1BuyY+YK01PEyijZez/VJNa/Zw8htsK63moV4X8oGlkO9g225cjfS2rFiWLzDdT1rz0JHZOo8HpgtqJG8bL0dvsNEz0h1T9ElSit4aBqzsfw2lSZ9iE2euAAAAAElFTkSuQmCC',
      status: 2,
      created_time: 1627977600,
    },
  ];

  afterAll(() => {
    jest.resetAllMocks();
  });

  describe('#rendering', () => {
    it('renders the table with offer data', async () => {
      renderApp(<OfferTable offerList={mockOffers} />);

      expect(screen.getByRole('table')).toMatchSnapshot();
    });

    it('renders a message when offerList is empty', async () => {
      renderApp(<OfferTable offerList={[]} />);

      expect(screen.getByText('No offers available')).toBeInTheDocument();
    });
  });

  describe('#error-handling', () => {
    it('displays "N/A" if OfferStatusMap key is missing', async () => {
      renderApp(<OfferTable offerList={[{ ...mockOffers[0], status: 'unknown' as unknown as OfferStatus }]} />);

      expect(screen.getByText('NA')).toBeInTheDocument();
    });
  });
});
