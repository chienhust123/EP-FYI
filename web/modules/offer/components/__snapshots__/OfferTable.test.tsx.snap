// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferTable #rendering renders the table with offer data 1`] = `
<table
  class="m_b23fa0ef mantine-Table-table"
  style="--table-vertical-spacing: calc(0.4375rem * var(--mantine-scale));"
>
  <thead
    class="m_b242d975 mantine-Table-thead"
  >
    <tr
      class="m_4e7aa4fd mantine-Table-tr"
      data-with-row-border="true"
    >
      <th
        class="m_4e7aa4f3 mantine-Table-th"
      >
        Company
      </th>
      <th
        class="m_4e7aa4f3 mantine-Table-th"
      >
        Location
      </th>
      <th
        class="m_4e7aa4f3 mantine-Table-th"
      >
        Position
      </th>
      <th
        class="m_4e7aa4f3 mantine-Table-th"
      >
        Total Package
      </th>
      <th
        class="m_4e7aa4f3 mantine-Table-th"
      >
        Status
      </th>
      <th
        class="m_4e7aa4f3 mantine-Table-th"
      />
    </tr>
  </thead>
  <tbody
    class="m_b2404537 mantine-Table-tbody"
  >
    <tr
      class="m_4e7aa4fd mantine-Table-tr"
      data-with-row-border="true"
    >
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        Tech Innovators
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        San Francisco
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        Software Engineer
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
         
        $120,000
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        Uploaded
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        <a
          href="/offer/detail?id=1"
        >
          <button
            class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
            data-variant="outline"
            style="--ai-bg: transparent; --ai-hover: var(--mantine-color-blue-outline-hover); --ai-color: var(--mantine-color-blue-outline); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
            type="button"
          >
            <span
              class="m_8d3afb97 mantine-ActionIcon-icon"
            >
              <svg
                class="tabler-icon tabler-icon-eye "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0"
                />
                <path
                  d="M21 12c-2.4 4 -5.4 6 -9 6c-3.6 0 -6.6 -2 -9 -6c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6"
                />
              </svg>
            </span>
          </button>
        </a>
      </td>
    </tr>
    <tr
      class="m_4e7aa4fd mantine-Table-tr"
      data-with-row-border="true"
    >
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        Global Finance
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        London
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        Financial Analyst
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
         
        Â£70,000
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        Approved
      </td>
      <td
        class="m_4e7aa4ef mantine-Table-td"
      >
        <a
          href="/offer/detail?id=2"
        >
          <button
            class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
            data-variant="outline"
            style="--ai-bg: transparent; --ai-hover: var(--mantine-color-blue-outline-hover); --ai-color: var(--mantine-color-blue-outline); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
            type="button"
          >
            <span
              class="m_8d3afb97 mantine-ActionIcon-icon"
            >
              <svg
                class="tabler-icon tabler-icon-eye "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0"
                />
                <path
                  d="M21 12c-2.4 4 -5.4 6 -9 6c-3.6 0 -6.6 -2 -9 -6c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6"
                />
              </svg>
            </span>
          </button>
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
